pkg/util/helper/work.go

```diff
@@ -82,7 +83,6 @@ func CreateOrUpdateWork(client client.Client, workMeta metav1.ObjectMeta, resour
 			if util.GetLabelValue(runtimeObject.Labels, workv1alpha2.WorkPermanentIDLabel) == "" {
 				runtimeObject.Labels = util.DedupeAndMergeLabels(runtimeObject.Labels, map[string]string{workv1alpha2.WorkPermanentIDLabel: uuid.New().String()})
 			}
-			util.RecordManagedLabels(runtimeObject)
```

Comment:
Oh, great find! The `workv1alpha2.ManagedLabels` should be recorded on the `workload` rather than the `work`.

Can you post a screenshot of your fixed test results?


LGTM

/cc @RainbowMango @XiShanYongYe-Chang PTAL.

Reply:
LGTM
thanks.

Reply:
> Can you post a screenshot of your fixed test results?

of course.
test label: `foo: bar`
![exam](https://github.com/karmada-io/karmada/assets/89241565/198b0222-5b22-4468-a914-7af036ca8eda)